src/v/cluster/members_manager.cc

```diff
@@ -53,7 +54,18 @@ members_manager::members_manager(
   , _allocator(allocator)
   , _storage(storage)
   , _as(as)
-  , _rpc_tls_config(config::shard_local_cfg().rpc_server_tls()) {}
+  , _rpc_tls_config(config::shard_local_cfg().rpc_server_tls())
+  , _update_queue(max_updates_queue_size) {
+    auto sub = _as.local().subscribe([this]() noexcept {
+        _update_queue.abort(
+          std::make_exception_ptr(ss::abort_requested_exception{}));
+    });
+    if (!sub) {
+        _update_queue.abort(
+          std::make_exception_ptr(ss::abort_requested_exception{}));
+    }
+    _queue_abort_subscription = std::move(*sub);
```

Comment:
is it safe to dereference `sub` if `!sub` is true?

Reply:
good point.